tools/rosconsole/src/rosconsole/logger_level_service_caller.py

```diff
@@ -0,0 +1,141 @@
+        """
+        set_logger_level_nodes = []
+        nodes = rosnode.get_node_names()
+        for name in sorted(nodes):
+            for service in rosservice.get_service_list(name):
+                if service == name + '/set_logger_level':
+                    set_logger_level_nodes.append(name)
+        return set_logger_level_nodes
+
+    def _refresh_loggers(self, node):
+        """
+        Stores a list of loggers available for passed in node
+        :param node: name of the node to query, ''str''
+        :raises: :exc:`ROSTopicException` If topic type cannot be determined or loaded
+        """
+        self._current_loggers = []
+        self._current_levels = {}
+        # Construct the service name, taking into account our namespace
+        servicename = rosgraph.names.ns_join(
+            rosgraph.names.ns_join(rosgraph.names.get_ros_namespace(), node),
+            'get_loggers')
+        # Construct the service name, taking into account our namespace
+        servicename = rosgraph.names.resolve_name(
+            servicename, rosgraph.names.get_ros_namespace())
+        try:
+            service = rosservice.get_service_class_by_name(servicename)
+        except rosservice.ROSServiceException as e:
+            raise ROSConsoleException("node %s doesn't exist or doesn't support query" % node)
+
+        request = service._request_class()
+        proxy = rospy.ServiceProxy(str(servicename), service)
+        try:
+            response = proxy(request)
+        except rospy.ServiceException as e:
+            raise ROSConsoleException("node %s logger request failed" % node)
+
+        if response._slot_types[0] == 'roscpp/Logger[]':
+            for logger in getattr(response, response.__slots__[0]):
+                self._current_loggers.append(getattr(logger, 'name'))
+                self._current_levels[getattr(logger, 'name')] = getattr(logger, 'level')
+        else:
+            raise ROSConsoleException(repr(response))
+
+    def send_logger_change_message(self, node, logger, level):
+        """
+        Sends a logger level change request to 'node'.
+        :param node: name of the node to chaange, ''str''
+        :param logger: name of the logger to change, ''str''
+        :param level: name of the level to change, ''str''
+        :returns: True if the response is valid, ''bool''
+        :returns: False if the request raises an exception or would not change the state, ''bool''
+        """
+        # Construct the service name, taking into account our namespace
+        servicename = rosgraph.names.ns_join(
+            rosgraph.names.ns_join(rosgraph.names.get_ros_namespace(), node),
+            'set_logger_level')
+        # Construct the service name, taking into account our namespace
+        servicename = rosgraph.names.resolve_name(
+            servicename, rosgraph.names.get_ros_namespace())
+        if self._current_levels[logger].lower() == level.lower():
```

Comment:
With the command line only accepting lower case input the `.lower()` can be removed on both sides.